external/vulkancts/modules/vulkan/pipeline/vktPipelineColorWriteEnableTests.cpp

```diff
@@ -1109,6 +1133,10 @@ void AddSingleTestCaseDynamic(
 
 	std::vector<tcu::Vec4> clearColorsPerAttachment(kNumColorAttachments, config.clearColorValue);
 
+	if (config.isReversed()) {
+		ApplyChannelMask(meshColorsPerAttachment, channelMask, config.clearColorValue);
+	}
+
 	config.expectedColor = MaskVector(clearColorsPerAttachment, meshColorsPerAttachment, mask, inverse);
```

Comment:
When config.isReversed() is false, the channelMask is no long applied to the expectedColor, causing the tests the fail. In other words, the final result should still have the channelMask applied.

Reply:
right so, if i understand this correctly, this assumption was false
> These tests assume that, when dynamically setting color write enables via `vkCmdSetColorWriteEnableEXT`, `colorWriteMask` has no bearing on behavior and the above logic is only relevant to a newly created pipeline.

`colorWriteMask`  is still taken into account when setting via `vkCmdSetColorWriteEnableEXT`

Reply:
Yeah it's the combination of the latest vkCmdBindPipeline static state and the latest vkCmdSetColorWriteEnableEXT dynamic state that is applied to the draw. A call to vkCmdSetColorWriteEnableEXT doesn't upset the active colorWriteMask state of the actively bound pipeline.

Reply:
Right that should be fixed now. Thank you!